docs/ai/conceptual/embeddings.md

```diff
@@ -0,0 +1,75 @@
+---
+title: "How Embeddings Extend Your AI Model's Reach"
+description: "Learn how embeddings extend the limits and capabilities of AI models in .NET."
+author: catbutler
+ms.topic: concept-article #Don't change.
+ms.date: 04/11/2024
+
+#customer intent: As a .NET developer, I want to understand how embeddings extend LLM limits and capabilities in .NET so that I have more semantic context and better outcomes for my AI apps.
+
+---
+
+# Embeddings in .NET
+
+This article explains how embeddings work in .NET.
+
+Embeddings are the way you express semantic meaning to an LLM. Essentially, embeddings are numeric representations of non-numeric data that preserve its semantic meaning. You can use embeddings to help an AI model understand the meaning of inputs so that it can perform comparisons and transformations, such as summarizing text or creating images from text descriptions. LLMs can use embeddings immediately, and you can store embeddings in vector databases to provide semantic memory for LLMs as-needed.
+
+Summarizing text is especially useful for [prompt engineering](/), because it gives you a way to use far fewer tokens to repesent a given text. You use instructions in prompt engineering to tell an LLM what to do, and can include text content with the instruction to give the model more context. However, LLMs limit the number of tokens per input, which constrains how much text you can include in prompts. If you try to include more than the limit in one input, the model will ignore some or all of that input. Some LLMs [charge per token](/pricing/details/cognitive-services/openai-service/) or have [quota systems](/azure/ai-services/openai/quotas-limits#regional-quota-limits), more reasons to watch your token counts. If you create an embedding for the text content you want to include, the LLM can find a shorter piece of text that has a similar vector, so you can use it instead of the original text to reduce the number of tokens you use to represent the content's meaning.
```

Comment:
If you create an embedding for the text content you want to include, the LLM can find a shorter piece of text that has a similar vector, so you can use it instead of the original text to reduce the number of tokens you use to represent the content's meaning.

Does the LLM use the embedding to generate a shorter text that still counts towards the token limit or does it just use the embedding in addition to the rest of prompt? In the latter case you'd only have to worry about the tokens for the part of the prompt you haven't already calculated an embedding for.

Reply:
It uses the embeddings to find a shorter, similar piece of text and tokenizes that. It uses vector analysis to compare embeddings (long text, candidate short text) and pick the best short text to tokenize in lieu of the long text.